src/main/scala/com/softwaremill/mqperf/Receiver.scala

```diff
@@ -52,26 +56,32 @@ class ReceiverRunnable(
       var lastReceivedNano = start
 
       while (System.nanoTime() - lastReceivedNano < timeoutNanos) {
-        val received = doReceive(mqReceiver, msgTimer)
+        val received = doReceive(mqReceiver, msgTimer, clusterLatencyTimer)
         if (received > 0) {
           val nowNano = System.nanoTime()
           lastReceivedNano = nowNano
-          meter.mark(received)
+          meter.mark()
         }
       }
       logger.info(s"Test finished, last message read $timeout ago")
-      TestMetrics.receive(metricRegistry).foreach(reportResults.report)
+      ReceiverMetrics(metricRegistry).foreach(reportResults.report)
     }
     finally {
       mqReceiver.close()
     }
   }
 
-  private def doReceive(mqReceiver: mq.MqReceiver, msgTimer: Timer) = {
+  private def doReceive(mqReceiver: mq.MqReceiver, msgTimer: Timer, clusterTimer: Timer) = {
     val before = System.nanoTime()
     val msgs = mqReceiver.receive(receiveMsgBatchSize)
     if (msgs.nonEmpty) {
       val after = System.nanoTime()
+      val afterMs = after / 1000000L
+      msgs.foreach {
+        case (_, msg) =>
+          val msgTimestamp = msg.substring(msg.indexOf('_') + 1).toLong
```

Comment:
Maybe I'll just change the way timestamp is serialized into message: msg = timestamp (as string) prepended with zeroes to fill up to required length in bytes. This way the "deserialization" would be just a `.toLong` call.